src/Chainweb/Pact/PactService.hs

```diff
@@ -445,8 +447,11 @@ validateChainwebTxsPreBlock dbEnv cp bh hash txs = do
         let pactHash = view P.cmdHash tx
         mb <- _cpLookupProcessedTx cp pactHash
         if mb == Nothing
-        then checkAccount tx
+        -- prop_tx_ttl_newblock_validate
+        then (&&) <$> checkAccount tx <*> return (k tx)
```

Comment:
Rather than making this single validator more complicated, keep it the same as before and compose single-purpose validations using a combinator

Reply:
done.